core/trino-main/src/test/java/io/trino/server/ui/TestMultipartUiCookie.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server.ui;
+
+import com.google.common.collect.ImmutableMap;
+import jakarta.ws.rs.core.Cookie;
+import jakarta.ws.rs.core.NewCookie;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class TestMultipartUiCookie
+{
+    private static final String COOKIE_NAME = "__UI_Cookie";
+
+    private static final MultipartUiCookie COOKIE = new MultipartUiCookie(COOKIE_NAME, "/location");
+
+    @Test
+    public void testCookieNames()
+    {
+        assertThat(COOKIE.cookieName(0)).isEqualTo(COOKIE_NAME);
+        assertThat(COOKIE.cookieName(1)).isEqualTo(COOKIE_NAME + "_1");
+        assertThat(COOKIE.cookieName(500)).isEqualTo(COOKIE_NAME + "_500");
+    }
+
+    @Test
+    public void testShortValueRoundTrip()
+    {
+        String longToken = "123456789".repeat(100); // < 4096 cookie value length limit
+        NewCookie[] newCookies = COOKIE.create(longToken, Instant.EPOCH, true);
+        assertThat(newCookies).hasSize(1);
+        assertThat(newCookies).extracting(NewCookie::getName).hasSameElementsAs(List.of(COOKIE.cookieName(0)));
+        assertThat(COOKIE.read(cookies(newCookies))).contains(longToken);
+
+        NewCookie[] deleteCookies = COOKIE.delete(cookies(newCookies), true);
+        assertThat(deleteCookies).hasSize(1);
+        assertThat(deleteCookies).extracting(NewCookie::getName).hasSameElementsAs(List.of(COOKIE.cookieName(0)));
```

Comment:
nit: Can we assert  value has `delete` ?

Reply:
THe delete value doesn't matter. MaxAge does matter

Reply:
In that case, can we assert the MaxAge ?